word_segmentation.py

```diff
@@ -205,140 +294,115 @@ def clean_line(line):
-                    icu_bies = get_bies(char_brkpoints, icu_word_brkpoints)
+                    # Computing BIES corresponding to the manually segmented and ICU segmented
+                    true_bies = get_bies(line.char_brkpoints, line.man_word_brkpoints)
+                    true_bies_str = get_bies_string_from_softmax(np.transpose(true_bies))
+                    icu_bies = get_bies(line.char_brkpoints, line.icu_word_brkpoints)
                     icu_bies_str = get_bies_string_from_softmax(np.transpose(icu_bies))
 
                     # Counting the number of mismatches between icu_bies and true_bies
                     icu_total_bies_lengths += len(icu_bies_str)
                     icu_mismatch += diff_strings(true_bies_str, icu_bies_str)
 
+                    # Computing the F1-score between icu_bies and true_bies
+                    f1_output = compute_f1_score(true_bies=true_bies_str, est_bies=icu_bies_str)
+                    correctly_segmented_words += f1_output[1]
+                    true_words += f1_output[2]
+                    segmented_words += f1_output[3]
+
                     # Demonstrate how icu segmenter works
-                    if demonstrate:
-                        char_segmented_str = get_segmented_string(unsegmented_line, char_brkpoints)
-                        print("Cat: {}, Text number: {}".format(cat, text_num))
-                        print("LINE {} - UNSEG LINE       : {}".format(line_counter, unsegmented_line))
-                        print("LINE {} - TRUE SEG LINE    : {}".format(line_counter, line))
-                        print("LINE {} - ICU SEG LINE     : {}".format(line_counter, icu_word_segmented_str))
-                        print("LINE {} - GRAPHEME CLUSTERS: {}".format(line_counter, char_segmented_str))
-                        print("LINE {} - TRUE BIES STRING : {}".format(line_counter, true_bies_str))
-                        print("LINE {} -  ICU BIES STRING : {}".format(line_counter, icu_bies_str))
-                        print("LINE {} - TRUE WORD BREAKS : {}".format(line_counter, word_brkpoints))
+                    if verbose:
+                        print("Cat: {}, Text number: {}, Line: {}".format(cat, text_num, line_counter))
+                        line.display()
+                        print("true bies string  : {}".format(true_bies_str))
+                        print("icu bies string   : {}".format(icu_bies_str))
                         print('**********************************************************************************')
-
                     line_counter += 1
 
-    icu_accuracy = 1 - icu_mismatch/icu_total_bies_lengths
+    icu_bies_accuracy = 1 - icu_mismatch/icu_total_bies_lengths
+
+    precision = correctly_segmented_words / segmented_words
+    recall = correctly_segmented_words / true_words
+    icu_f1_accuracy = 0
+    if precision + recall != 0:
+        icu_f1_accuracy = 2 * precision * recall / (precision + recall)
```

Comment:
I'm not super happy about having the F1 score computation logic in two different places (here and in the main compute_f1_score)